cranelift-codegen/src/licm.rs

```diff
@@ -156,12 +155,25 @@ fn trivially_unsafe_for_licm(opcode: Opcode) -> bool {
         || opcode.writes_cpu_flags()
 }
 
+fn is_unsafe_load(inst_data: &InstructionData) -> bool {
+    match *inst_data {
+        InstructionData::Load { flags, .. } | InstructionData::LoadComplex { flags, .. } => {
+            !flags.readonly() || !flags.notrap()
+        }
+        _ => inst_data.opcode().can_load(),
+    }
+}
+
```

Comment:
This is now the same code as simple_gvn's `is_load_and_not_readonly`. I'm not particular on which name we use, but at this point it'd be good to factor this function out into a utility function in cranelift-codegen/src/ir/instructions.rs. It looks like we could move `trivially_unsafe_for_licm`/`trivially_unsafe_for_gvn` there too.

Reply:
(Oops, I originally had a duplicate of this comment here.)

Reply:
Actually, it's not the same code.  Here I also check !notrap(), which gvn does not do (and probably should not need to do).

Reply:
Agreed.